mfs.go

```diff
@@ -0,0 +1,370 @@
+package shell
+
+import (
+	"context"
+	"io"
+
+	files "github.com/ipfs/go-ipfs-files"
+)
+
+type MfsLsEntry struct {
+	Name string
+	Type uint8
+	Size uint64
+	Hash string
+}
+
+type filesLsOutput struct {
+	Entries []*MfsLsEntry
+}
+
+type filesFlushOutput struct {
+	Cid string
+}
+
+type filesStatOutput struct {
+	Blocks         int
+	CumulativeSize uint64
+	Hash           string
+	Local          bool
+	Size           uint64
+	SizeLocal      uint64
+	Type           string
+	WithLocality   bool
+}
+
+type filesOpt func(*RequestBuilder) error
+type filesLs struct{}
+type filesChcid struct{}
+type filesMkdir struct{}
+type filesRead struct{}
+type filesWrite struct{}
+type filesStat struct{}
+
+var (
+	FilesLs    filesLs
+	FilesChcid filesChcid
+	FilesMkdir filesMkdir
+	FilesRead  filesRead
+	FilesWrite filesWrite
+	FilesStat  filesStat
+)
+
+// Long use long listing format
+func (filesLs) Long(long bool) filesOpt {
+	return func(rb *RequestBuilder) error {
+		rb.Option("long", long)
+		return nil
+	}
+}
+
+// U do not sort; list entries in directory order
+func (filesLs) U(u bool) filesOpt {
```

Comment:
This option is only relevant on the command line.

Reply:
OK，I'll remove it. BTW, it would be better to add a note in the API document [here](https://docs.ipfs.io/reference/http/api/#api-v0-files-ls).

Reply:
Unfortunately, those docs are auto-generated from the comandline help text. We'll need to add a field to commands for "api notes" (we need this anyways, honestly).

Reply:
Yes, I opened an issue to track this here: https://github.com/ipfs/http-api-docs/issues/31